### PR TITLE
Cleanup ListControl tests

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
Was going to write some test for subclasses of `ListControl` but noticed the tests were a bit messy - not disposing/using `WinFormsFact` etc. so went through and cleaned up the test class

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2566)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListControlTests.cs [hunk 1]
```diff
@@ -250,21 +260,24 @@ public void BindingContext_SetWithDataSourceWithBindingContext_GetReturnsExpecte
...
-        [Fact]
-        public void BindingContext_SetWithDataSourceAndDisplayMemberWithBindingContext_GetReturnsExpected()
+        [WinFormsFact]
+        public void ListControl_BindingContext_SetWithDataSourceAndDisplayMemberWithBindingContext_GetReturnsExpected()
         {
             var originalValue = new BindingContext();
             var value = new BindingContext();
             var dataSource = new List<DataClass>();
-            var control = new SubListControl
+            using var control = new SubListControl
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListControlTests.cs]: Just curious, why didn't you add a test for `IsHandleCreated` in this test?

Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListControlTests.cs]: Oversight! Fixed

**Comment by hughbe** [comment]
Not really, because each of validation is different in subtle ways and we'd end up having corner cases, if statements and the like (increasing cyclomatic complexity and making the tests harder to read/understand/update as we'd have endless `VerifyXYZ` methods)

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListControlTests.cs [hunk 2]
```diff
@@ -2501,299 +2591,266 @@ public void ListControl_GetStyle_Invoke_ReturnsExpected(ControlStyles flag, bool
...
             };
-
+        
+            // Call with handler.
             control.Format += handler;
             control.OnFormat(eventArgs);
             Assert.Equal(1, callCount);
-
-            // Should not call if the handler is removed.
+
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListControlTests.cs]: please remove the extra spaces
